tests/ci/cpu_unit_tests.yml

```diff
@@ -0,0 +1,101 @@
+# cpu_unit_tests.yml
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+#
+#
+# NOTE: start with cpu to make sure everything still works
+#
+# use variable group name
+variables:
+- group: AzureKeyVaultSecrets
+- name  : 'test'
+  value : 'tests/ci/run_pytest.py'
+- name  : 'testfolder'
+  value : './tests/unit'
+- name  : 'testmarkers'
+  value : '"not notebooks and not spark and not gpu"'
+- name  : 'junitxml'
+  value : '"reports/test-unit.xml"'
+#   
+- name  : 'maxnodes'
+  value : 4
+#- name  : 'rg'
+#  value : 'recommender'
+#- name  : 'wsname'
+#  value : 'RecoWS'
+- name  : 'reponame'
+  value : 'Recommenders'
+- name  : 'branch'
+  value : 'azure-pipelines-bz'
+- name  : 'clustername'
+  value : 'Reco-cpu-test'
+# GPU
+#- name  : 'vmsize'
+#  value : 'STANDARD_NC6'
+# CPU
+- name  : 'vmsize'
+  value : 'STANDARD_D3_V2'
+- name  : 'dockerproc'
+  value : 'cpu'
+- name  : 'expname'
+  value : 'cpu_unit_tests'
+
+#trigger:
+#- azure-pipelines-bz
+# 
+
+pr:
+- staging 
+- master
+  
+pool:
+ vmImage: 'ubuntu-16.04'
+
+steps:
+
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '3.6'
+    architecture: 'x64'
+  displayName: 'Use Python 3.6'
+
+- script: | 
+   az login --service-principal -u $(ClientID) -p $(ClientSecret) --tenant $(TenantID)
+   
+  displayName: 'Login to Azure'
+
+- bash: |
+    echo "##vso[task.prependpath]/data/anaconda/bin"
+  displayName: Add Conda to PATH
+
+- script: 
+     pip install azureml-sdk
+  continueOnError: true
+  displayName: 'install azureml-sdk'
+  
+- script: 
+     python scripts/generate_conda_file.py --name reco 
+  displayName: ' generate_conda_file.py'
+
+- script: |
+     chmod +x scripts/*.py
+     chmod +x tests/ci/*.py
+  displayName: 'ls'
+
+- script: |

```

Comment:
why not combine this with azureml-sdk pip install?

Reply:
I put the chmod in it's own task so it's easier to know what went wrong if something hiccups. If they are separate, it's trivial to know the issue. If they are together, it means reading the pipeline logs and those are not as kind.